File: Fundamentals of Discrete Mathematics/Combinаtorics/Java/task_5.java
import java.util.ArrayList;
import java.util.Scanner;

public class task_5 {
    public static void generateCombinations(ArrayList<Integer> combination, int n, int k, int offset) {
        if (k == 0) {
            for (int i : combination) {
                System.out.print(i + " ");
            }
            System.out.println();
            return;
        }

        for (int i = offset; i <= n - k; i++) {
            combination.add(i);
            generateCombinations(combination, n, k - 1, i + 1);
            combination.remove(combination.size() - 1);
        }
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int k = scanner.nextInt();
        int n = scanner.nextInt();

        ArrayList<Integer> combination = new ArrayList<>();
        generateCombinations(combination, n, k, 0);
    }
}

File: Fundamentals of Discrete Mathematics/Combinаtorics/C++/task_5.cpp
#include <iostream>
#include <vector>

using namespace std;

void generate_combinations(vector<int>& combination, int n, int k, int offset) {
    if (k == 0) {
        for (int i : combination) {
            cout << i << " ";
        }
        cout << endl;
        return;
    }

    for (int i = offset; i <= n - k; i++) {
        combination.push_back(i);
        generate_combinations(combination, n, k - 1, i + 1);
        combination.pop_back();
    }
}

int main() {
    int n, k;
    cin >> k >> n;

    vector<int> combination;
    generate_combinations(combination, n, k, 0);

    return 0;
}